test/test_browser.py

```diff
@@ -5382,7 +5382,7 @@ def test_full_js_library_strict(self):
     'minimal_runtime_pthreads_and_closure': (['-sMINIMAL_RUNTIME', '-pthread', '--closure', '1', '-Oz'],),
     'pthreads_es6': (['-pthread', '-sPTHREAD_POOL_SIZE=2', '-sEXPORT_ES6'],),
     'es6': (['-sEXPORT_ES6'],),
-    'strict': (['-sSTRICT'],),
+    'strict': (['-sSTRICT', '-sINCOMING_MODULE_JS_API=canvas,instantiateWasm,monitorRunDependencies,onAbort,onExit,print,setStatus,wasm,wasmMemory'],),
```

Comment:
Why is this change needed for this PR?

Reply:
I was running the audio tests with `-sSTRICT` and couldn't understand how any of them ran without it. They build, they don't run, e.g., compile any example with `STRICT` and run it:

```
Uncaught RuntimeError: Aborted(`Module.wasmMemory` was supplied but `wasmMemory` not included in INCOMING_MODULE_JS_API)
    at abort (index.js:592:11)
    at ignoredModuleProp (index.js:823:5)
    at checkIncomingModuleAPI (index.js:1548:3)
    at index.js:223:1
```

It's for sure separate, so can remove it and come back to the tests later.

Reply:
If wasmMemory is requires by for AUDIO_WORKLET then its should be added explicitly in link.py, so that its always part of `INCOMING_MODULE_JS_API` (even in strict mode).

Reply:
I created #22971 and I'll come back to this, since it's not new.

Reply:
Great.  I'm happy to take a look at that one too if you like.  Or I can help you fix.  Should be fairly simple.

Reply:
I took a quick look and I'm a missing some of the concepts. Feel free to take a look or just throw some hints my way. I won't be able to look at it until next week now.

I stalled after a quick look because I wasn't sure things like `canvas` should be automatically added, so wanted to test first. But the locking was higher on the todo list (and I was hoping to get #22949 solved and in, which I think I'll just write because exposing sample rate, etc., in an extensible way is pretty important).

Plus we'd like to ship our code built around this at some point! 😀

Reply:
Resolving this and will come back to ##22971.